bn_mp_factors_sort.c

```diff
@@ -0,0 +1,56 @@
+#include "tommath_private.h"
+#ifdef BN_MP_FACTORS_SORT_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis
+ *
+ * LibTomMath is a library that provides multiple-precision
+ * integer arithmetic as well as number theoretic functionality.
+ *
+ * The library was designed directly after the MPI library by
+ * Michael Fromberger but has been written from scratch with
+ * additional optimizations in place.
+ *
+ * SPDX-License-Identifier: Unlicense
+ */
+
+
+/* Sort factor-array (an array of mp_int's) */
+#ifdef LTM_USE_EXTRA_FUNCTIONS
+/* TODO: Sorts inline. Work on copy instead? */
+int mp_factors_sort(mp_factors *f)
+{
+   int i, idx, e = MP_OKAY;
+   mp_int tmp;
+
+   if ((e = mp_init(&tmp)) != MP_OKAY) {
+      return e;
+   }
+
+   /*
+      It will be almost always be already sorted and even if it is unsorted
+      it will just show a bushy tail. So insertion sort it is.
+   */
+   for (i = 1 ; i < f->length; i++) {
+      idx = i;
+      while ((idx > 0) && (mp_cmp(&(f->factors[idx-1]),&(f->factors[idx])) == MP_GT)) {
+         if ((e = mp_copy(&(f->factors[idx]), &tmp)) != MP_OKAY) {
```

Comment:
we're deep inside the library, why should we do a deep copy of all `mp_int` elements where it would be sufficient to exchange the `mp_int`?

Reply:
Another leftover. I always start with `mp_copy` and check later if it can be replaced with `mp_exch`.

Thanks for taking the time to go through the whole mess!
(Lunchtime is a bit short today, the whole thing needs to wait till tonight.)